src/main/java/frc/team2412/robot/commands/indexer/IndexBitmapCommand.java

```diff
@@ -102,87 +104,91 @@ public IndexBitmapCommand(final IndexerSubsystemSuperStructure indexerMotorSubsy
 
 	@Override
 	public void execute() {
-		IntakeDirection intakeDirection = RobotState.getintakeDirection();
-		final boolean intakeOn = (intakeDirection != IntakeDirection.NONE);
-		if (intakeDirection == IntakeDirection.NONE) {
+		IntakeDirection realIntakeDirection = RobotState.getintakeDirection();
+		final IntakeDirection intakeDirection;
+		final boolean intakeOn = (realIntakeDirection != IntakeDirection.NONE);
+		if (realIntakeDirection == IntakeDirection.NONE) {
 			if (s_lastIntakeDirection == IntakeDirection.BACK) {
 				intakeDirection = IntakeDirection.BACK;
 			} else {
 				intakeDirection = IntakeDirection.FRONT;
 			}
 		} else {
-			s_lastIntakeDirection = intakeDirection;
+			intakeDirection = realIntakeDirection;
+			s_lastIntakeDirection = realIntakeDirection;
 		}
 
-		int sensorBitmap = 0;
-		switch (intakeDirection) {
-		case BOTH:
-		case FRONT:
-			sensorBitmap = (m_indexerSubsystem.getIndexerSensorSubsystem().getSensorBitmapFrontLSB()
-					& VALID_SENSOR_BITS);
-			break;
-		case BACK:
-			sensorBitmap = (m_indexerSubsystem.getIndexerSensorSubsystem().getSensorBitmapBackLSB()
-					& VALID_SENSOR_BITS);
-			break;
-		case NONE:
-			assert (false);
-			break;
-		}
+		int sensorBitmap = getSensorBitmap(intakeDirection);
 
-		IndexCommandEntry indexCommand = null;
-		for (final IndexCommandEntry entry : IndexCommandEntry.values()) {
-			if (entry.expectedBits == (sensorBitmap & entry.validBits)) {
-				indexCommand = entry;
-			}
-		}
+		Arrays.stream(IndexCommandEntry.values()).filter(c -> c.expectedBits == (sensorBitmap & c.validBits)).findAny()
```

Comment:
While this is nice, if we were to show this to someone else and try to explain it, it might be difficult. The other version of this reads a lot better. Is there a way to merge the two?

Reply:
it uses the stream api, which you should look into, it turns into more functional programming